apps/src/lib/client/tx.rs

```diff
@@ -512,6 +513,79 @@ pub async fn submit_init_validator<'a>(
             .or_else(|| Some(config.ledger.chain_id.clone())),
         ..tx_args.clone()
     };
+
+    // Check that the address is established
+    if !address.is_established() {
+        edisplay_line!(
+            namada.io(),
+            "The given address {address} is not established. Only established \
+             address can become a validator.",
+        );
+        if !tx_args.force {
+            safe_exit(1)
+        }
+    };
+
+    // Check that the address is not already a validator
+    if rpc::is_validator(namada.client(), &address).await {
+        edisplay_line!(
+            namada.io(),
+            "The given address {address} is already a validator",
+        );
+        if !tx_args.force {
+            safe_exit(1)
+        }
+    };
+
+    // If the address is not yet a validator, it cannot have self-bonds, but it
+    // may have delegations. It has to unbond those before it can become a
+    // validator.
+    if rpc::has_bonds(namada.client(), &address).await {
```

Comment:
If the established address then goes and unbonds all of its delegations, with this logic it will be able to immediately then become a validator successfully before waiting the unbonding period, since `unbond` will decrement the bonds to 0 immediately. Just want to check that this is the desired behavior or if we want to wait the unbonding period (which would require `has_bonds` to check existing `unbonds` as well)

Reply:
but this check is only in the transaction creation code, no? we'd need this check to be in the tx verification logic later (maybe it is)

Reply:
I found the other check in the code later. I think not checking unbonds is fine in principle, we should have a quick test for this though to make sure nothing breaks.

Reply:
Yeah, I think this is safe. Only thing that can be done with `unbonds` is to withdraw them once they're ready. If the delegation gets slashed, then only the withdrawal action will affected. Rewards are not auto-bonded so they wouldn't affect this either (they have to be claimed back into the account first)